### PR TITLE
Setup the User Schema #3 && Setup Post Schema #4

### LINKED ISSUE #3: Setup the User Schema
Make sure to setup the Models needed for the project, following the database schema we created

### LINKED ISSUE #4: Setup Post Schema
Setup the Model needed for Post using the database scheam we created

### PR DESCRIPTION
**Pull request by khaldarov**
Added both User Schema and Post Schema (thank you [@okurezgi](https://github.com/okurezgi)). Fixes #3 and #4.

### CODE REVIEW

**Review by awiednoor — COMMENTED** [review]
It all looks good with some minor changes:
let's change these names to make them more descriptive
in **constants.js**
instead of _status_ in line 13 let's make it _reportStatus_
and let's change conditions to _overused, underused_
in **post.js**
let's change _requestedUser_ to _requestUser_
in **user.js** 
let's change _hash_ to be _passwordhash_

A question for **user.js** file
shouldn't  _Donated, Requested, Received_ refer to post schema, what are they referring to here ?

**Comment by khaldarov** [comment]
For the question, I tried to comply with the DB schema. It says that asked fields should only have the post's id.

**Review by awiednoor — APPROVED** [review]

File: src/lib/constants.js [hunk 1]
```diff
@@ -1,3 +1,38 @@
 module.exports = Object.freeze({
-  CONSTANT: 'some value',
+  // models' constants start
```
Comment (halitbatur) [inline src/lib/constants.js]: can we delete this comment? it does not add any new information

File: src/lib/constants.js [hunk 2]
```diff
@@ -1,3 +1,38 @@
...
+  enumShippingOptions: [
+    'Meet up',
+    'Drop off',
+    'Free shipping',
+    'Paid shipping',
+    'To be determined',
+  ],
+  enumPostType: ['Request', 'Donate'],
+  enumPostStatus: ['Published', 'Verified'],
+  // models' constants end
```
Comment (halitbatur) [inline src/lib/constants.js]: this one too

File: src/lib/constants.js [hunk 3]
```diff
@@ -1,3 +1,38 @@
 module.exports = Object.freeze({
-  CONSTANT: 'some value',
+  // models' constants start
+  enumGender: ['Female', 'Male', 'Prefer not to say', 'Other'],
+  enumNationality: [
```
Comment (halitbatur) [inline src/lib/constants.js]: for constants maybe we can use ALL_CAPITAL

**Comment by khaldarov** [comment]
Fixed requested revision. Added missing auth files for /register, /login, and /logout endpoints; added them to `app.js`

**Review by halitbatur — APPROVED** [review]

**Pull request merged**